reposerver/repository/repository.go

```diff
@@ -164,6 +164,13 @@ type operationSettings struct {
 	allowConcurrent bool
 }
 
+type operationContext struct {
```

Comment:
Wrote up some comments for these two types while reviewing, how does this sound:
```go
// operationContext contains request values which are generated by runRepoOperation (on demand) by a call to the
// provided operationContextSrc function.
type operationContext struct {

	// application path or helm chart path
	appPath string

	// output of 'git verify-(tag/commit)', if signature verifiction is enabled (otherwise "")
	verificationResult string
}

// The 'operation' function parameter of 'runRepoOperation' may call this function to retrieve
// the appPath or GPG verificationResult.
// Failure to generate either of these values will return an error which may be cached by
// the calling function (for example, 'runManifestGen')
type operationContextSrc = func() (*operationContext, error)
```

Reply:
thank you @jgwest ! It looks great! Added to the PR.